python_fundamentals-master/07_classes_objects_methods/07_02_shapes.py

```diff
@@ -9,27 +9,25 @@
 '''
 
 class rectangle:
+    square = True                   #class variable
     def __init__(self, length, width):
-        self.length = length
+        self.length = length        #attributes (variables that are only defined in init)
         self.width = width
-    def area(self):
         self.area = self.length * self.width
-        return print(f"the area of the rectangle is {self.area}")
-    def perimeter(self):
-        self.perimeter()  = 2  * (self.width + self.length)
-        return print(f"the perimeter is the rectangle is {self.perimeter}")
+        self.perimeter = 2 * (self.width + self.length)
+    def __str__(self):
+        return f"the area of the rectangle is {self.area} and the perimeter is the rectangle is {self.perimeter}"
 
 rec_1 = rectangle(5, 2)
-print(rec_1.area())
-print(rec_1.perimeter())
-
+print(rec_1)
 
 class circle:
-    def __init__(self, radius):
+    def __init__(self, radius, area, circumference):
         self.radius = radius
-    def area(self, length, width):
-        self.area = length * width
-        return print(f"the area of the circle is {self.area}")
-    def circumference(self, r):
-        self.circumference() = 2 * 3,1415 * self.radius
-        return print(f"the circumference of the circle is {self.circumference()}")
\ No newline at end of file
+        self.area = 3.1415 * self.radius * self.radius
```

Comment:
I don't think this is the correct equation right?

Reply:
I changed it to pi * radius ** 2 -- > although pi * radius *  radius is also possible right?